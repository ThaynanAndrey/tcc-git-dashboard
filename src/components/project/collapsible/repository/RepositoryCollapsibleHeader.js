import React from 'react';

const styleCollapsibleHeader = {
    display: "flex",
    justifyContent: "space-between"
};

const styleCollapsibleHeaderText = {
    display: "flex"
};

/**
 * Stateless component with the function of presenting a Header Collapsible to Repositories.
 * 
 * @author Thaynan Nunes
 */
const RepositoryCollapsibleHeader = ({ idProject, redirectPage }) => (
    <div className="collapsible-header" style={styleCollapsibleHeader}>
        <div style={styleCollapsibleHeaderText}>
            <i className="material-icons">dehaze</i>
            Reposit√≥rios
        </div>
        <div>
            <button className="btn-small waves-effect waves-light green darken-2"
                    type="button" name="cadastrar-pr"
                    onClick={() => redirectPage(`${idProject}/adicionarRepositorios`)}>
                Adicionar
                <i className="material-icons left">add</i>
            </button>
        </div>
    </div>
);

export default RepositoryCollapsibleHeader;