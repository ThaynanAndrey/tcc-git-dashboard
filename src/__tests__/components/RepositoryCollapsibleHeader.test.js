import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import RepositoryCollapsibleHeader from '../../components/project/collapsible/repository/RepositoryCollapsibleHeader';

describe('Test RepositoryCollapsibleHeader', () => {
  it('renders correctly a component to show RepositoryCollapsibleHeader', () => {
    const tree = mount(
        <RepositoryCollapsibleHeader />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders correctly a component to show RepositoryCollapsibleHeader with correct props', () => {
    const idProject = "123";
    const redirectPage = jest.fn();

    const tree = mount(
        <RepositoryCollapsibleHeader
            idProject={idProject}
            redirectPage={redirectPage}
        />
    );

    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.prop('idProject')).toEqual(idProject);
    expect(tree.prop('redirectPage')).toEqual(redirectPage);
  });

  it('renders correctly a component to show RepositoryCollapsibleHeader with correct classes', () => {
    const idProject = "123";
    const redirectPage = jest.fn();

    const tree = mount(
        <RepositoryCollapsibleHeader
            idProject={idProject}
            redirectPage={redirectPage}
        />
    );

    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.find(".collapsible-header").exists()).toBeTruthy();
    expect(tree.find('.btn-small.waves-effect.waves-light.green.darken-2').exists()).toBeTruthy();
  });
})