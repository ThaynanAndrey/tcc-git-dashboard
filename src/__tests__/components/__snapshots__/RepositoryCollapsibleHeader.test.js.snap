// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RepositoryCollapsibleHeader renders correctly a component to show RepositoryCollapsibleHeader 1`] = `
<RepositoryCollapsibleHeader>
  <div
    className="collapsible-header"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <i
        className="material-icons"
      >
        dehaze
      </i>
      Repositórios
    </div>
    <div>
      <button
        className="btn-small waves-effect waves-light green darken-2"
        name="cadastrar-pr"
        onClick={[Function]}
        type="button"
      >
        Adicionar
        <i
          className="material-icons left"
        >
          add
        </i>
      </button>
    </div>
  </div>
</RepositoryCollapsibleHeader>
`;

exports[`Test RepositoryCollapsibleHeader renders correctly a component to show RepositoryCollapsibleHeader with correct classes 1`] = `
<RepositoryCollapsibleHeader
  idProject="123"
  redirectPage={[MockFunction]}
>
  <div
    className="collapsible-header"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <i
        className="material-icons"
      >
        dehaze
      </i>
      Repositórios
    </div>
    <div>
      <button
        className="btn-small waves-effect waves-light green darken-2"
        name="cadastrar-pr"
        onClick={[Function]}
        type="button"
      >
        Adicionar
        <i
          className="material-icons left"
        >
          add
        </i>
      </button>
    </div>
  </div>
</RepositoryCollapsibleHeader>
`;

exports[`Test RepositoryCollapsibleHeader renders correctly a component to show RepositoryCollapsibleHeader with correct props 1`] = `
<RepositoryCollapsibleHeader
  idProject="123"
  redirectPage={[MockFunction]}
>
  <div
    className="collapsible-header"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <i
        className="material-icons"
      >
        dehaze
      </i>
      Repositórios
    </div>
    <div>
      <button
        className="btn-small waves-effect waves-light green darken-2"
        name="cadastrar-pr"
        onClick={[Function]}
        type="button"
      >
        Adicionar
        <i
          className="material-icons left"
        >
          add
        </i>
      </button>
    </div>
  </div>
</RepositoryCollapsibleHeader>
`;
